scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -577,7 +577,7 @@ impl TomlManifest {
 
         if targets.is_empty() {
             trace!("manifest has no targets, assuming default `lib` target");
-            let default_source_path = root.join(DEFAULT_SOURCE_PATH);
+            let default_source_path = root.join(DEFAULT_SOURCE_PATH.as_path());
```

Comment:
FYI 

```suggestion
            let default_source_path = root.join(&DEFAULT_SOURCE_PATH);
```

or

```suggestion
            let default_source_path = root.join(*DEFAULT_SOURCE_PATH);
```

I don't know which one is correct

Reply:
None works for me 😱 
```
   --> scarb/src/core/manifest/toml_manifest.rs:580:50
    |
580 |             let default_source_path = root.join(&DEFAULT_SOURCE_PATH);
    |                                            ----  ^^^^^^^^^^^^^^^^^^^ the trait `std::convert::AsRef<camino::Utf8Path>` is not implemented for `once_cell::sync::Lazy<camino::Utf8PathBuf>`
    |                                            |
    |                                            required by a bound introduced by this call
    |
    = note: required for `&once_cell::sync::Lazy<camino::Utf8PathBuf>` to implement `std::convert::AsRef<camino::Utf8Path>`
```

```
error[E0507]: cannot move out of dereference of `once_cell::sync::Lazy<camino::Utf8PathBuf>`
   --> scarb/src/core/manifest/toml_manifest.rs:580:49
    |
580 |             let default_source_path = root.join(*DEFAULT_SOURCE_PATH);
    |                                                 ^^^^^^^^^^^^^^^^^^^^ move occurs because value has type `camino::Utf8PathBuf`, which does not implement the `Copy` trait
```